src/custom_modules/authorization/authMiddleware.ts

```diff
@@ -33,11 +35,21 @@ export const auth = (permission, api = true) => {
   return [checkJwt, requiredScopes(permission)];
 };
 
-export const checkDonor = (donorId, req, res, next) => {
-  const handler = claimCheck(
-    (claims) =>
-      (claims as ExtendedJWTPayload)[roleClaim].includes("admin") ||
-      claims[useridClaim] === donorId
+const userIsAdmin = (claims: JWTPayload) => {
+  const roleClaims = claims[roleClaim];
+  if (!Array.isArray(roleClaims)) return false;
+  return roleClaims[roleClaim].includes("admin");
```

Comment:
Can't have the `ExtendedJWTPayload` anymore as the keys are now dynamic, but the type safety of this should be higher even as it doesn't rely on type assertions